ynr/apps/official_documents/models.py

```diff
@@ -20,13 +20,17 @@ class OfficialDocument(TimeStampedModel):
         (NOMINATION_PAPER, "Nomination paper", "Nomination papers"),
     )
 
+    PDF = models.FileField(
+        null=True, blank=True, validators=[FileExtensionValidator(["pdf"])]
+    )
+
     document_type = models.CharField(
         blank=False,
         choices=[(d[0], d[1]) for d in DOCUMENT_TYPES],
         max_length=100,
     )
     uploaded_file = models.FileField(
-        upload_to=document_file_name, max_length=800
+        upload_to=document_file_name, max_length=800, validators=PDF.validators
```

Comment:
Why would this affect file types for photo uploads? https://app.circleci.com/pipelines/github/DemocracyClub/yournextrepresentative/1405/workflows/75f30fa3-600c-43c6-b3b4-a4384ab26f67/jobs/1416 This test passes if I change the example photo from '.jpg' to '.pdf'.

Reply:
I was thinking that we add the validation on the upload form rather than the model. As if we force it on the model, then all the PDF conversion work you have done will never be used (also these changes are actually adding a new `PDF` field to the model but I dont really understand why?). So I would revert these changes, and then on the [UploadDocumentForm add](https://github.com/DemocracyClub/yournextrepresentative/blob/a6e768ec64b23eaede3eba7f021a86e741b70598/ynr/apps/official_documents/forms.py#L6):
```
uploaded_file = forms.FileField(validators=[FileExtensionValidator(allowed_extensions=["pdf"])])
```
This then means that when we download the SOPN's from the spreadsheet we can still download docx etc and try to convert them, but on the candidates website it will display an error message if someone tried to upload a non-PDF (you can try this in your browser).

As for the failing test, if you look at the [full test](https://github.com/DemocracyClub/yournextrepresentative/blob/master/ynr/apps/official_documents/tests/test_upload.py#L77) you can see it is testing the SOPN upload form not the photo uploader. I think the confusion comes because the test is using an example image jpg rather than a file. So it makes sense this test now fails because we are adding validation to only accept `.pdf` extensions. So I think it is fine for you to change the file extension used to `.pdf` and it will pass.

Reply:
Ok, this all makes much more sense. I initially added the validation to the `UploadDocumentForm` but then couldn't get the structure right. Thank you!